### DISCUSSION TITLE
How to make a writable nested relationship?

### DISCUSSION THREAD

**Discussion by ghost** [django-json-api/django-rest-framework-json-api#1160] [Q&A] [answered]
If I do something similar to this https://www.django-rest-framework.org/api-guide/serializers/#writable-nested-representations,

using the following Serializer:

```
class TestSerializer(serializers.ModelSerializer):
    profile = ProfileSerializer()

class Meta:
    model = Test
```

It expects my data to be structured like this:

```
...
"relationships": {
                    "profile": {
                        "data": {
                                "first_name": "John",
                                "last_name": "Test"
                        }
                    }
                }
...
```

instead of like this:

```
...
"relationships": {
                    "profile": {
                        "data": {
                            "type": "Profile",
                            "attributes": {
                                "first_name": "John",
                                "last_name": "Test"
                            }
                        }
                    }
                }
...
```

could someone explain how to make a writable nested relationship with valid JSON:API schema?

**Comment by jokiefer** [comment, marked as answer]
Take a look at the [json:api specs](https://jsonapi.org/format/#document-resource-object-linkage). Both examples are not json:api specific.

Nested creation is not a well supported thing in json:api. Only [creating](https://jsonapi.org/format/#crud-creating) single resources is covered by the specs.

Maybe [atomic operations](https://jsonapi.org/ext/atomic/) would be help you to create resources in a single transaction. But this extension is not implemented in this package.

**Reply by jokiefer** [reply]
Here is a first alpha version for an [atomic operations extension]( https://github.com/jokiefer/drf-json-api-atomic-operations). Maybe this could help you a bit.

**Reply by ghost** [reply]
Thanks for the info! I'll follow the atomic operations implementation, but I think I'll go for some extra post requests in the short term.
